scripts/variantstore/wdl/ImportGenomes.wdl

```diff
@@ -185,3 +228,131 @@ task CreateImportTsvs {
+
+      # runtime
+      Int? preemptible_tries
+      String docker
+    }
+
+  command <<<
+    set -x
+    set -e
+
+    DIR="~{storage_location}/~{datatype}_tsvs/"
+
+    for TABLE_ID in $(seq 1 ~{max_table_id}); do
+      PARTITION_STRING=""
+      if [ ~{partitioned} == "true" ]; then
+        let "PARTITION_START=(${TABLE_ID}-1)*4000+1"
+        let "PARTITION_END=$PARTITION_START+3999"
+        let "PARTITION_STEP=1"
+        PARTITION_FIELD="sample_id"
+        PARTITION_STRING="--range_partitioning=$PARTITION_FIELD,$PARTITION_START,$PARTITION_END,$PARTITION_STEP"
+      fi
+
+      printf -v PADDED_TABLE_ID "_%03d" ${TABLE_ID}
+      FILES="~{datatype}${PADDED_TABLE_ID}_*"
+
+      NUM_FILES=$(gsutil ls $DIR$FILES | wc -l)
+
+      # create the table
+      PREFIX=""
+      if [ -n "~{uuid}" ]; then
+          PREFIX="~{uuid}_"
+      fi
+
+      if [ $NUM_FILES -gt 0 ]; then
+        if [ ~{numbered} != "true" ]; then
+          PADDED_TABLE_ID=""  #override table id to empty string, but it is needed to get the files
+        fi
+
+        TABLE="~{dataset_name}.${PREFIX}~{datatype}${PADDED_TABLE_ID}"
+
+        # Check that the table has not been created yet
+        set +e
+        bq show --project_id ~{project_id} $TABLE > /dev/null
+        BQ_SHOW_RC=$?
+        set -e
+        if [ $BQ_SHOW_RC -ne 0 ]; then
+          echo "making table $TABLE"
+          bq --location=US mk ${PARTITION_STRING} --project_id=~{project_id} $TABLE ~{schema}
+        fi
+
+        echo "$TABLE,$DIR,$FILES" >> table_dir_files.csv
+      else
+        echo "no ${FILES} files to process"
+      fi
+    done
+  >>>
+
+  output {
+    Array[String] table_dir_files_list = read_lines("table_dir_files.csv")
+  }
+
+  runtime {
+    docker: docker
+    memory: "3 GB"
+    disks: "local-disk 10 HDD"
+    preemptible: select_first([preemptible_tries, 5])
+    cpu: 1
+  }
+}
+
+task LoadTable {
+  meta {
+    volatile: true
+  }
+
+  input {
+    String table_dir_files_str
+    String project_id
+    File schema
+    String load
+    Array[String] tsv_creation_done
+
+    Int? preemptible_tries
+    String docker
+  }
+
+  command <<<
+    TABLE=$(echo ~{table_dir_files_str} | cut -d, -f1)
+    DIR=$(echo ~{table_dir_files_str} | cut -d, -f2)
+    FILES=$(echo ~{table_dir_files_str} | cut -d, -f3)
+
+    #load should be false if using Google Storage Transfer so that the tables will be created by this script, but no data will be uploaded.
+    if [ ~{load} = true ]; then
+      bq load --location=US --project_id=~{project_id} --skip_leading_rows=1 --source_format=CSV -F "\t" $TABLE $DIR$FILES ~{schema} || exit 1
+      echo "ingested ${FILES} file from $DIR into table $TABLE"
+      gsutil mv $DIR$FILES ${DIR}done/
+    else
+      echo "${FILES} will be ingested from $DIR by Google Storage Transfer"
+    fi
+  >>>
+
+  runtime {
+    docker: docker
+    memory: "3 GB"
+    disks: "local-disk 10 HDD"
+    preemptible: select_first([preemptible_tries, 5])
```

Comment:
Interesting... what happens if we are in the midst of a big data import `bq load` and we get pre-empted?  My guess is the load still happens... but then we retry and load the data a second time...

Reply:
turn preemptibles off for now